tests/core/test_composite.py

```diff
@@ -439,7 +440,8 @@ def test_transform_filter(ant, sphere, airplane, tetbeam, inplace):
     keys_after = output.keys()
 
     assert (output is multi) == inplace
-    assert [(block_in is block_out) == inplace for block_in, block_out in zip(multi, output)]
+    for i, _ in enumerate(multi):
+        assert (multi[i] is output[i]) == inplace or (multi[i] is None)
```

Comment:
```suggestion
    for block_in, block_out in zip(multi, output):
        assert (block_in is block_out) == inplace or (block_in is None)
```

Reply:
Using zip like this won't work for MultiBlock because of #6600. So we have to iterate without zip, e.g. with `enumerate` so that __getitem__ is used internally instead of __next__

Reply:
```suggestion
    # https://github.com/pyvista/pyvista/pull/6599/files#r1739931261
    for i, _ in enumerate(multi):
        assert (multi[i] is output[i]) == inplace or (multi[i] is None)
```

Reply:
ref #6644